### PR TITLE
Stop noisy modifications

### PR DESCRIPTION
**Pull request by johnbelamaric**
### 1. What does this pull request do?

Stops continuous changes to the serial number due to irrelevant endpoints changes.

### 2. Which issues (if any) are related?

#1516 

### 3. Which documentation changes (if any) need to be made?

None.

### CODE REVIEW

**Comment by johnbelamaric** [comment]
After this change:

```
ANN-M-JBELAMARIC:~ jbelamaric$ while [ 1 ]; do dig soa -p 10053 @localhost cluster.local | grep SOA; sleep 2; done
;cluster.local.			IN	SOA
cluster.local.		300	IN	SOA	ns.dns.cluster.local. hostmaster.cluster.local. 1518203650 7200 1800 86400 30
;cluster.local.			IN	SOA
cluster.local.		300	IN	SOA	ns.dns.cluster.local. hostmaster.cluster.local. 1518203650 7200 1800 86400 30
;cluster.local.			IN	SOA
cluster.local.		300	IN	SOA	ns.dns.cluster.local. hostmaster.cluster.local. 1518203650 7200 1800 86400 30
;cluster.local.			IN	SOA
cluster.local.		300	IN	SOA	ns.dns.cluster.local. hostmaster.cluster.local. 1518203650 7200 1800 86400 30
;cluster.local.			IN	SOA
cluster.local.		300	IN	SOA	ns.dns.cluster.local. hostmaster.cluster.local. 1518203650 7200 1800 86400 30
;cluster.local.			IN	SOA
cluster.local.		300	IN	SOA	ns.dns.cluster.local. hostmaster.cluster.local. 1518203650 7200 1800 86400 30
;cluster.local.			IN	SOA
cluster.local.		300	IN	SOA	ns.dns.cluster.local. hostmaster.cluster.local. 1518203650 7200 1800 86400 30
;cluster.local.			IN	SOA
cluster.local.		300	IN	SOA	ns.dns.cluster.local. hostmaster.cluster.local. 1518203650 7200 1800 86400 30
^C
```

**Comment by johnbelamaric** [comment]
needs tests...

File: plugin/kubernetes/controller.go [hunk 1]
```diff
@@ -431,6 +431,67 @@ func (dns *dnsControl) updateModifed() {
...
+	if o, ok := objOld.(*api.Endpoints); ok {
+		n := newObj.(*api.Endpoints)
+		if endpointsEquivalent(o, n) {
+			return
+		}
+	}
+	dns.updateModifed()
+}
+
+// checks if the update to an endpoint is something
```
Comment (miekg) [inline plugin/kubernetes/controller.go]: It's not exported, but it still good form to start the comment with `endpointsEquivalent` (just endpontsEqual to mimic std lib naming?)

Comment (johnbelamaric) [reply plugin/kubernetes/controller.go]: Ok. They are not equal in the normal sense. They just produce the same DNS entries.

**Comment by johnbelamaric** [comment]
fixes #1517

File: plugin/kubernetes/xfr_test.go [hunk 2]
```diff
@@ -109,3 +111,92 @@ func difference(testRRs []dns.RR, gotRRs []dns.RR) []dns.RR {
...
+	epG := api.Endpoints{
+		Subsets: []api.EndpointSubset{{
+			Addresses: []api.EndpointAddress{{IP: "1.2.3.4", Hostname: "foo"}},
+			Ports:     []api.EndpointPort{{Name: "http", Port: 80, Protocol: "TCP"}},
+		}},
+	}
+	epH := api.Endpoints{
+		Subsets: []api.EndpointSubset{{
+			Addresses: []api.EndpointAddress{{IP: "1.2.3.4", Hostname: "foo"}},
+			Ports:     []api.EndpointPort{{Name: "thhp", Port: 80, Protocol: "TCP"}},
```
Comment (bradbeam) [inline plugin/kubernetes/xfr_test.go]: `thhp` is this intentional?

**Review by bradbeam — APPROVED** [review]

**Review by chrisohaver — APPROVED** [review]

**Pull request merged**